File: JavaScript/8js 3/sumMul.js
/*
|--------------------------------------------------------------------------
| Sum of Multiples
|--------------------------------------------------------------------------
|
*/

function sumMul(n, m) {
   let res = 0;
   for (let i = n; i < m; i += n) res += i;
   return res != 0 ? res : "INVALID";
}


function sumMul(n, m) {
   return (~~(m / n) * (m + n - (m % n))) / 2 || "INVALID";
}


function sumMul(n, m) {
   return m <= n ? "INVALID" : Math.floor(m / n) * (Math.floor(m / n) + 1) * (n / 2);
}


console.log(sumMul(0, 0), "INVALID");
console.log(sumMul(2, 9), 20);
console.log(sumMul(4, -7), "INVALID");

File: Python/8py 3/sum_mul.py
"""
|--------------------------------------------------------------------------
| Sum of Multiples
|--------------------------------------------------------------------------
|
"""

def sum_mul(n, m):
   return "INVALID" if m <= 0 or n <= 0 else sum([i for i in range(n,m,n)])


def sum_mul(n, m):
   return sum(range(n, m, n)) if n > 0 and m > 0 else "INVALID"


def sum_mul(n, m):
   return sum([x for x in range(n, m, n)]) if n > 0 < m else "INVALID"


print(sum_mul(0, 0), 'INVALID')
print(sum_mul(2, 9), 20)
print(sum_mul(3, 13), 30)
print(sum_mul(4, 123), 1860)
print(sum_mul(581, 256), 0)
print(sum_mul(0, 2), 'INVALID')
print(sum_mul(-5, 5), 'INVALID')